launcher/InstanceList.cpp

```diff
@@ -286,39 +259,84 @@ bool InstanceList::isGroupCollapsed(const QString& group)
     return m_collapsedGroups.contains(group);
 }
 
+bool InstanceList::trashInstance(const InstanceId& id)
+{
+    auto inst = getInstanceById(id);
+    if (!inst) {
+        qDebug() << "Cannot trash instance" << id << ". No such instance is present (deleted externally?).";
+        return false;
+    }
+
+    auto cachedGroupId = m_instanceGroupIndex[id];
+
+    if (m_instanceGroupIndex.remove(id)) {
+        saveGroupList();
+    }
```

Comment:
This would remove the group association, even if trashing failed below.

Reply:
`deleteInstance` also has this problem...